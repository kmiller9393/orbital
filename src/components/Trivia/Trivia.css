.trivia-header-image {
  height: 55px;
  margin-top: 1.5rem;
  width: 125px; }

.trivia-header {
  color: #f7f7f7;
  margin-top: 2.5rem; }

.trivia-container {
  font-family: 'Chalkboard SE', 'Times New Roman', serif;
  align-items: center;
  display: flex;
  flex-direction: column; }

.trivia-button {
  background-color: #009c92;
  border-color: #009c92;
  border-radius: 5px;
  height: 45px;
  margin-bottom: 1rem;
  width: 175px; }

.trivia-page-link {
  color: #f7f7f7;
  font-size: 1.3rem;
  text-decoration: none; }

.trivia-button-container {
  align-items: center;
  display: flex;
  flex-direction: column;
  margin-top: 1.5rem; }
