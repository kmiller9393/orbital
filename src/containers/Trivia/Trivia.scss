.loading-gif {
  max-height: 50px;
  max-width: 50px;
}

.trivia-header-image {
  height: 55px;
  margin-top: 1.5rem;
  width: 125px;
}

.trivia-header {
  color: #f7f7f7;
  margin-top: 2.5rem;
}

.trivia-header-container {
  padding-right: 6.5rem;
}

.trivia-container {
  align-items: center;
  display: flex;
  flex-direction: column;
  font-family: 'Chalkboard SE', 'Times New Roman', serif;
  height: 100%;
}

.center {
  justify-content: center;
}

.back-arrow {
  height: 40px;
  padding: 0rem 4rem 0.5rem 0rem;
}

.trivia-button {
  background-color: #009c92;
  border-color: #009c92;
  border-radius: 5px;
  box-shadow: 0px 8px 15px rgba(0, 0, 0, 0.3);
  cursor: pointer;
  height: 45px;
  margin-bottom: 1rem;
  width: 175px;
}

.trivia-button:active {
  background-color: #02b0a4;
}

.trivia-page-link {
  color: #f7f7f7;
  font-size: 1.3rem;
  text-decoration: none;
}

.trivia-button-container {
  align-items: center;
  display: flex;
  flex-direction: column;
  margin-top: 2.5rem;
}
