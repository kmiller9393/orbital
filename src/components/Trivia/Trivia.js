import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import orbital from '../../images/orbital-header.svg';
import './Trivia.css';

export default class Trivia extends Component {
  render() {
    return (
      <div className="trivia-container">
        <NavLink to="/">
          <img className="trivia-header-image" src={orbital} alt="Orbital" />
        </NavLink>
        <h2 className="trivia-header">Trivia Center</h2>
        <div className="trivia-button-container">
          <button className="trivia-button">
            <NavLink className="trivia-page-link" to="/trivia-one" name="1">
              Week 1
            </NavLink>
          </button>
          <button className="trivia-button">
            <NavLink className="trivia-page-link" to="/trivia-two" name="2">
              Week 2
            </NavLink>
          </button>
          <button className="trivia-button">
            <NavLink className="trivia-page-link" to="/trivia-three" name="3">
              Week 3
            </NavLink>
          </button>
          <button className="trivia-button">
            <NavLink className="trivia-page-link" to="/trivia-four" name="4">
              Week 4
            </NavLink>
          </button>
          <button className="trivia-button">
            <NavLink className="trivia-page-link" to="/trivia-five" name="5">
              Week 5
            </NavLink>
          </button>
        </div>
      </div>
    );
  }
}
