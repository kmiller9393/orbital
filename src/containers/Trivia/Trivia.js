import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { populateTriviaFacts } from '../../thunks';
import PropTypes from 'prop-types';
import orbital from '../../images/orbital-header.svg';
import './Trivia.css';

class Trivia extends Component {
  setTriviaInfo = async () => {
    await this.props.populateTrivia();
  };

  render() {
    return (
      <div className="trivia-container">
        <NavLink to="/">
          <img className="trivia-header-image" src={orbital} alt="Orbital" />
        </NavLink>
        <h2 className="trivia-header">Trivia Center</h2>
        <div className="trivia-button-container">
          <button className="trivia-button" onClick={this.setTriviaInfo}>
            <NavLink className="trivia-page-link" to="/trivia-one">
              Week 1
            </NavLink>
          </button>
          <button className="trivia-button">
            <NavLink className="trivia-page-link" to="/trivia-two">
              Week 2
            </NavLink>
          </button>
          <button className="trivia-button">
            <NavLink className="trivia-page-link" to="/trivia-three">
              Week 3
            </NavLink>
          </button>
          <button className="trivia-button">
            <NavLink className="trivia-page-link" to="/trivia-four">
              Week 4
            </NavLink>
          </button>
          <button className="trivia-button">
            <NavLink className="trivia-page-link" to="/trivia-five">
              Week 5
            </NavLink>
          </button>
        </div>
      </div>
    );
  }
}

export const mapDispatchToProps = dispatch => ({
  populateTrivia: () => dispatch(populateTriviaFacts())
});

export default connect(
  null,
  mapDispatchToProps
)(Trivia);

const { func } = PropTypes;
Trivia.propTypes = {
  populateTrivia: func
};
