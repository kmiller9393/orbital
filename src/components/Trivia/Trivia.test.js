import { mapDispatchToProps } from './Trivia';

describe('Trivia', () => {
  describe('Trivia component', () => {});
});
