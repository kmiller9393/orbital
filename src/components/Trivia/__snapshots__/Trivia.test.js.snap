// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trivia should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Trivia />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavLink
          activeClassName="active"
          aria-current="page"
          to="/"
>
          <img
                    alt="Orbital"
                    className="trivia-header-image"
                    src="orbital-header.svg"
          />
</NavLink>,
        <h2
          className="trivia-header"
>
          Trivia Center
</h2>,
        <div
          className="trivia-button-container"
>
          <button
                    className="trivia-button"
          >
                    <NavLink
                              activeClassName="active"
                              aria-current="page"
                              className="trivia-page-link"
                              name="1"
                              to="/trivia-one"
                    >
                              Week 1
                    </NavLink>
          </button>
          <button
                    className="trivia-button"
          >
                    <NavLink
                              activeClassName="active"
                              aria-current="page"
                              className="trivia-page-link"
                              name="2"
                              to="/trivia-two"
                    >
                              Week 2
                    </NavLink>
          </button>
          <button
                    className="trivia-button"
          >
                    <NavLink
                              activeClassName="active"
                              aria-current="page"
                              className="trivia-page-link"
                              name="3"
                              to="/trivia-three"
                    >
                              Week 3
                    </NavLink>
          </button>
          <button
                    className="trivia-button"
          >
                    <NavLink
                              activeClassName="active"
                              aria-current="page"
                              className="trivia-page-link"
                              name="4"
                              to="/trivia-four"
                    >
                              Week 4
                    </NavLink>
          </button>
          <button
                    className="trivia-button"
          >
                    <NavLink
                              activeClassName="active"
                              aria-current="page"
                              className="trivia-page-link"
                              name="5"
                              to="/trivia-five"
                    >
                              Week 5
                    </NavLink>
          </button>
</div>,
      ],
      "className": "trivia-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": <img
            alt="Orbital"
            className="trivia-header-image"
            src="orbital-header.svg"
/>,
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Orbital",
            "className": "trivia-header-image",
            "src": "orbital-header.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Trivia Center",
          "className": "trivia-header",
        },
        "ref": null,
        "rendered": "Trivia Center",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="trivia-button"
>
              <NavLink
                            activeClassName="active"
                            aria-current="page"
                            className="trivia-page-link"
                            name="1"
                            to="/trivia-one"
              >
                            Week 1
              </NavLink>
</button>,
            <button
              className="trivia-button"
>
              <NavLink
                            activeClassName="active"
                            aria-current="page"
                            className="trivia-page-link"
                            name="2"
                            to="/trivia-two"
              >
                            Week 2
              </NavLink>
</button>,
            <button
              className="trivia-button"
>
              <NavLink
                            activeClassName="active"
                            aria-current="page"
                            className="trivia-page-link"
                            name="3"
                            to="/trivia-three"
              >
                            Week 3
              </NavLink>
</button>,
            <button
              className="trivia-button"
>
              <NavLink
                            activeClassName="active"
                            aria-current="page"
                            className="trivia-page-link"
                            name="4"
                            to="/trivia-four"
              >
                            Week 4
              </NavLink>
</button>,
            <button
              className="trivia-button"
>
              <NavLink
                            activeClassName="active"
                            aria-current="page"
                            className="trivia-page-link"
                            name="5"
                            to="/trivia-five"
              >
                            Week 5
              </NavLink>
</button>,
          ],
          "className": "trivia-button-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                aria-current="page"
                className="trivia-page-link"
                name="1"
                to="/trivia-one"
>
                Week 1
</NavLink>,
              "className": "trivia-button",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "Week 1",
                "className": "trivia-page-link",
                "name": "1",
                "to": "/trivia-one",
              },
              "ref": null,
              "rendered": "Week 1",
              "type": [Function],
            },
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                aria-current="page"
                className="trivia-page-link"
                name="2"
                to="/trivia-two"
>
                Week 2
</NavLink>,
              "className": "trivia-button",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "Week 2",
                "className": "trivia-page-link",
                "name": "2",
                "to": "/trivia-two",
              },
              "ref": null,
              "rendered": "Week 2",
              "type": [Function],
            },
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                aria-current="page"
                className="trivia-page-link"
                name="3"
                to="/trivia-three"
>
                Week 3
</NavLink>,
              "className": "trivia-button",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "Week 3",
                "className": "trivia-page-link",
                "name": "3",
                "to": "/trivia-three",
              },
              "ref": null,
              "rendered": "Week 3",
              "type": [Function],
            },
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                aria-current="page"
                className="trivia-page-link"
                name="4"
                to="/trivia-four"
>
                Week 4
</NavLink>,
              "className": "trivia-button",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "Week 4",
                "className": "trivia-page-link",
                "name": "4",
                "to": "/trivia-four",
              },
              "ref": null,
              "rendered": "Week 4",
              "type": [Function],
            },
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                aria-current="page"
                className="trivia-page-link"
                name="5"
                to="/trivia-five"
>
                Week 5
</NavLink>,
              "className": "trivia-button",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "Week 5",
                "className": "trivia-page-link",
                "name": "5",
                "to": "/trivia-five",
              },
              "ref": null,
              "rendered": "Week 5",
              "type": [Function],
            },
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="active"
            aria-current="page"
            to="/"
>
            <img
                        alt="Orbital"
                        className="trivia-header-image"
                        src="orbital-header.svg"
            />
</NavLink>,
          <h2
            className="trivia-header"
>
            Trivia Center
</h2>,
          <div
            className="trivia-button-container"
>
            <button
                        className="trivia-button"
            >
                        <NavLink
                                    activeClassName="active"
                                    aria-current="page"
                                    className="trivia-page-link"
                                    name="1"
                                    to="/trivia-one"
                        >
                                    Week 1
                        </NavLink>
            </button>
            <button
                        className="trivia-button"
            >
                        <NavLink
                                    activeClassName="active"
                                    aria-current="page"
                                    className="trivia-page-link"
                                    name="2"
                                    to="/trivia-two"
                        >
                                    Week 2
                        </NavLink>
            </button>
            <button
                        className="trivia-button"
            >
                        <NavLink
                                    activeClassName="active"
                                    aria-current="page"
                                    className="trivia-page-link"
                                    name="3"
                                    to="/trivia-three"
                        >
                                    Week 3
                        </NavLink>
            </button>
            <button
                        className="trivia-button"
            >
                        <NavLink
                                    activeClassName="active"
                                    aria-current="page"
                                    className="trivia-page-link"
                                    name="4"
                                    to="/trivia-four"
                        >
                                    Week 4
                        </NavLink>
            </button>
            <button
                        className="trivia-button"
            >
                        <NavLink
                                    activeClassName="active"
                                    aria-current="page"
                                    className="trivia-page-link"
                                    name="5"
                                    to="/trivia-five"
                        >
                                    Week 5
                        </NavLink>
            </button>
</div>,
        ],
        "className": "trivia-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": <img
              alt="Orbital"
              className="trivia-header-image"
              src="orbital-header.svg"
/>,
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Orbital",
              "className": "trivia-header-image",
              "src": "orbital-header.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Trivia Center",
            "className": "trivia-header",
          },
          "ref": null,
          "rendered": "Trivia Center",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="trivia-button"
>
                <NavLink
                                activeClassName="active"
                                aria-current="page"
                                className="trivia-page-link"
                                name="1"
                                to="/trivia-one"
                >
                                Week 1
                </NavLink>
</button>,
              <button
                className="trivia-button"
>
                <NavLink
                                activeClassName="active"
                                aria-current="page"
                                className="trivia-page-link"
                                name="2"
                                to="/trivia-two"
                >
                                Week 2
                </NavLink>
</button>,
              <button
                className="trivia-button"
>
                <NavLink
                                activeClassName="active"
                                aria-current="page"
                                className="trivia-page-link"
                                name="3"
                                to="/trivia-three"
                >
                                Week 3
                </NavLink>
</button>,
              <button
                className="trivia-button"
>
                <NavLink
                                activeClassName="active"
                                aria-current="page"
                                className="trivia-page-link"
                                name="4"
                                to="/trivia-four"
                >
                                Week 4
                </NavLink>
</button>,
              <button
                className="trivia-button"
>
                <NavLink
                                activeClassName="active"
                                aria-current="page"
                                className="trivia-page-link"
                                name="5"
                                to="/trivia-five"
                >
                                Week 5
                </NavLink>
</button>,
            ],
            "className": "trivia-button-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="trivia-page-link"
                  name="1"
                  to="/trivia-one"
>
                  Week 1
</NavLink>,
                "className": "trivia-button",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "aria-current": "page",
                  "children": "Week 1",
                  "className": "trivia-page-link",
                  "name": "1",
                  "to": "/trivia-one",
                },
                "ref": null,
                "rendered": "Week 1",
                "type": [Function],
              },
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="trivia-page-link"
                  name="2"
                  to="/trivia-two"
>
                  Week 2
</NavLink>,
                "className": "trivia-button",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "aria-current": "page",
                  "children": "Week 2",
                  "className": "trivia-page-link",
                  "name": "2",
                  "to": "/trivia-two",
                },
                "ref": null,
                "rendered": "Week 2",
                "type": [Function],
              },
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="trivia-page-link"
                  name="3"
                  to="/trivia-three"
>
                  Week 3
</NavLink>,
                "className": "trivia-button",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "aria-current": "page",
                  "children": "Week 3",
                  "className": "trivia-page-link",
                  "name": "3",
                  "to": "/trivia-three",
                },
                "ref": null,
                "rendered": "Week 3",
                "type": [Function],
              },
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="trivia-page-link"
                  name="4"
                  to="/trivia-four"
>
                  Week 4
</NavLink>,
                "className": "trivia-button",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "aria-current": "page",
                  "children": "Week 4",
                  "className": "trivia-page-link",
                  "name": "4",
                  "to": "/trivia-four",
                },
                "ref": null,
                "rendered": "Week 4",
                "type": [Function],
              },
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="trivia-page-link"
                  name="5"
                  to="/trivia-five"
>
                  Week 5
</NavLink>,
                "className": "trivia-button",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "aria-current": "page",
                  "children": "Week 5",
                  "className": "trivia-page-link",
                  "name": "5",
                  "to": "/trivia-five",
                },
                "ref": null,
                "rendered": "Week 5",
                "type": [Function],
              },
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
